Improve Kibana dashboard and the screenshots in doc

zyiou opened the pull request:
This commit improves network policy diagram by grouping and color-coding flows using network policy. It shows the flows with no network policy in pie chart to provide complete information.

![flow-visualization-np-1](https://user-images.githubusercontent.com/36459546/109721376-1c7d9080-7b60-11eb-895a-2be985d800d1.png)

It also updates the screenshots of Kibana dashboard with better sankey diagrams (more variety). Example:
![flow-visualization-pod-to-pod-1](https://user-images.githubusercontent.com/36459546/109721444-34551480-7b60-11eb-9140-d1761cf0536d.png)

antoninbas reviewed (dismissed):
LGTM, any noteworthy change in kibana.ndjson?

zyiou commented:
> LGTM, any noteworthy change in kibana.ndjson?

Thanks! As I mentioned in description. In Kibana.ndjson we add color-coding by network policies and show flows with no network policies in pie chart.

srikartati reviewed (commented):
In the network policy snapshots, color-coded groups are not consistent with colors on vertical ingress and egress lines. Is this expected? If expected please specify what they correspond to for avoiding any confusion.

zyiou commented:
> In the network policy snapshots, color-coded groups are not consistent with colors on vertical ingress and egress lines. Is this expected? If expected please specify what they correspond to for avoiding any confusion.

Yes, it is expected. That is to distinguish source and destination pod of each flow. If they are using the same color, Kibana will treat flows under the same network policy as a whole that we cannot tell the throughput of single flow.

srikartati commented:
> > In the network policy snapshots, color-coded groups are not consistent with colors on vertical ingress and egress lines. Is this expected? If expected please specify what they correspond to for avoiding any confusion.
> 
> Yes, it is expected. That is to distinguish source and destination pod of each flow. If they are using the same color, Kibana will treat flows under the same network policy as a whole that we cannot tell the throughput of single flow.

I feel users could use Pod-to-Pod flow dashboard to figure out the throughput IMO. In network policy dashboard, if feasible, we can use the same color code for all the flows involving the same network policy even on the left vertical line. If not feasible, please add the above clarification in the document, where we add the note of network policy color coding.

zyiou commented:
> > > In the network policy snapshots, color-coded groups are not consistent with colors on vertical ingress and egress lines. Is this expected? If expected please specify what they correspond to for avoiding any confusion.
> > 
> > 
> > Yes, it is expected. That is to distinguish source and destination pod of each flow. If they are using the same color, Kibana will treat flows under the same network policy as a whole that we cannot tell the throughput of single flow.
> 
> I feel users could use Pod-to-Pod flow dashboard to figure out the throughput IMO. In network policy dashboard, if feasible, we can use the same color code for all the flows involving the same network policy even on the left vertical line. If not feasible, please add the above clarification in the document, where we add the note of network policy color coding.

As discussed offline, we decided to use egress network policy as source and ingress network policy as destination in sankey diagram. The updated visualization is as follows:
![flow-visualization-np-1](https://user-images.githubusercontent.com/36459546/110396170-5fd67400-8024-11eb-9774-da58c0874807.png)
![flow-visualization-np-2](https://user-images.githubusercontent.com/36459546/110396235-81cff680-8024-11eb-9baf-afecef070b7a.png)

srikartati commented on docs/network-flow-visibility.md:
```diff
@@ -457,24 +457,26 @@ Visualization Flow Record Dashboard">
 
 Network Policy dashboard provides filters over ingress network policy name and namespace, egress
-network policy name and namespace to view corresponding flow throughput under network policy.
+network policy name and namespace to view corresponding flow throughput under network policy. Flows
+are grouped and color-coded by network policies in the sankey diagram. When hovering over the flow,
+it will show corresponding Pod-to-Pod traffic details and network policies.
```
I think this description of color coding for network policy should be modified.

srikartati reviewed (commented):
Thanks for making the changes. Modified Sankey diagrams look good. 
One clarification: the color of flow pipes is based on source and destination tuple and that is why they are unique?

zyiou commented:
> Thanks for making the changes. Modified Sankey diagrams look good.
> One clarification: the color of flow pipes is based on source and destination tuple and that is why they are unique?

Thanks for pointing out. Updated the flow pipe color to be align with source. Do you think it is better?
![flow-visualization-np-1](https://user-images.githubusercontent.com/36459546/110401453-71bd1480-802e-11eb-9175-9dc5bb2d8626.png)
![flow-visualization-np-2](https://user-images.githubusercontent.com/36459546/110401459-7386d800-802e-11eb-9331-7071a5ba6308.png)

srikartati commented:
> > Thanks for making the changes. Modified Sankey diagrams look good.
> > One clarification: the color of flow pipes is based on source and destination tuple and that is why they are unique?
> 
> Thanks for pointing out. Updated the flow pipe color to be align with source. Do you think it is better?

Yes, I think this looks better and IMO users could easily distinguish network policies and flows going through them.

srikartati reviewed (approved):
LGTM

srikartati commented:
/skip-all
